plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/S3BlobStore.java

```diff
@@ -186,6 +193,18 @@ public static StorageClass initStorageClass(String storageClassStringValue) {
         return storageClass;
     }
 
+    public Tagging softDeleteTagging() {
+        Tag tag = Tag.builder()
+            .key(SOFT_DELETE_KEY.get(repositoryMetadata.settings()))
+            .value(SOFT_DELETE_VALUE.get(repositoryMetadata.settings()))
+            .build();
+        return Tagging.builder().tagSet(tag).build();
+    }
+
+    boolean softDeleteEnabled() {
+        return SOFT_DELETE_ENABLED.get(repositoryMetadata.settings());
```

Comment:
Just curious on why we are relying on a setting rather than a overridden method that can supply is with this information whether the soft deletion capability exists or not. With this we need to handle so many combinations of the actual support and passed settings which can probably overcomplicate things and will require us to handle them in code without a real need to do so.